templates/_macros/utils.twig

```diff
@@ -66,3 +66,135 @@
             height="{{ image.height }}">
     {% endif %}
 {% endmacro %}
+
+{#
+ |------------------------------------------------------------------------------
+ | Pagination for search
+ |------------------------------------------------------------------------------
+#}
+{% macro pagination(pageInfo, nearbyPagesLimit = 4) %}
+    {% set total = pageInfo.totalPages %}
+    {% set current = pageInfo.currentPage %}
+    {% spaceless %}
+        {% if total > 1 %}
+            <div class="row text-center">
+                <ul class="pagination">
+
+                {# Pagination Previous Button #}
+                {% if pageInfo.prevUrl %}
+                    <li><a href="{{ pageInfo.prevUrl }}" rel="prev">Previous</a></li>
+                {% endif %}
+
+                {% for i in 1..total %}
+                    {% if 0 == (current - nearbyPagesLimit) - loop.index %}
+                        <li><a href="{{ pageInfo.getPageUrl( 1 ) }}">1</a></li>
+                        {% if 1 != loop.index %}
+                            <li><span>…</span></li>
+                        {% endif %}
+                    {% elseif 0 == (current + nearbyPagesLimit) - loop.index  and (current + nearbyPagesLimit) < total %}
+                        <li><span>…</span></li>
+                    {% elseif 0 < (current - nearbyPagesLimit) - loop.index %}
```

Comment:
I don't understand this structure. If this statement is true, nothing happens? Same with the other one below? Why not fallback to the else directly. I think this section needs to be commented a little bit more. If you have more than 1-2 else if, you may want to change to a [switch statement](https://docs.craftcms.com/v3/dev/tags/switch.html).

Reply:
I've put lots of comment why I did it like that. Hopefully this will make it clearer. We can chat about it when you tested it.